docs/application/vscode-ext/tools/diagnostics.md

```diff
@@ -0,0 +1,142 @@
+     ![gcdump PerfView analysis view](media/dotnet-gcdump-perfview.png)
+
+  - Another way to open the **gcdump** file is to right click the gcdump file in **File Explorer Panel** of VS Code and select **Tizen: Open in Diagnostics File Viewer**. It will open the file in **PerfView** tool as seen below:
+     ![gcdump PerfView analysis view](media/dotnet-gcdump-open.png)
+
+    > [!NOTE]
+    > The above two options to open/view gcdump file is supported only in Windows OS.
+
+
+### dotnet-trace
+
+#### Step 1. Run dotnet-trace tool
+   - If no Tizen device is connected and no Tizen Emulators are running, then launch the type of Emulator you want to use for diagnosing your application:
+
+     ![Launch Emulator](media/start_emulator.png)
+
+   - Open the **Command Palette** and select **Tizen.NET: Run dotnet diagnostics tools**:
+
+     ![Run dotnet-trace](media/dotnet-diagnostics.png)
+
+   - Now select **trace** from the drop down list:
+
+     ![Select dotnet-trace](media/dotnet-trace-select.png)
+    
+   - Now select **profile** from the drop down list:
+
+     ![Select profile dotnet-trace](media/dotnet-trace-profile.png)
+
+   - Now select **format** from the drop down list:
+
+     ![Select format dotnet-trace](media/dotnet-trace-format.png)
+
+   - If everything is okay then the application starts, and it also starts collecting the trace file:
+
+     ![Tizen application running](media/dotnet-trace-running.png)
+
+   - Press **Cancel** button above to stop and generate the trace file.
+
+   - Once the trace file is generated, a message popup is shown to inform the user, and the file is added to the **trace** folder inside the **TizenDiagnostics** folder of the project:  
+
+     ![Completed dotnet-trace](media/dotnet-trace-completed.png)
+
+  
+
+#### Step 2. Analyze the trace file
+   - There are different tools for different **Format** to view the trace diagnostics data:
+   - **nettrace**
+       - Once the trace file is collected, a message popup is shown with the option to open the generated trace file. Once we click the **YES** option. It will open the file in **PerfView** tool as seen below:
+       ![trace PerfView analysis view](media/dotnet-trace-perfview.png)
+       - Right click the trace (.nettrace) file in **File Explorer Panel** of VS Code, and then select **Tizen: Open in Diagnostics File Viewer**. It will open the file in **PerfView** tool as seen below:
+       ![trace PerfView analysis view](media/dotnet-trace-open.png)
+       - > [!NOTE]
+         > The above two options to open/view nettrace file is supported only in Windows OS.
+   - **speedscope**
+       - Right click the trace (.nettrace) file in **File Explorer Panel** of VS code and select **Tizen: Open in Diagnostics File Viewer**. It will open the file in **SpeedScope** web tool as seen below:
```

Comment:
```suggestion
       - Right click the trace (.nettrace) file in **File Explorer Panel** of VS Code, and then select **Tizen: Open in Diagnostics File Viewer**. It will open the file in **SpeedScope** web tool as seen below:
```